js/app.js

```diff
@@ -1,32 +1,221 @@
-var warper = {
-  init: function() {
-    var endpoint = 'warper api endpoint';
-
-    var request = $.ajax({
-      url: endpoint,
-      method: 'GET'
-    });
-
-    request.done(function() {
-      // init
-    });
-
-    request.fail(function() {
-      warper.alert('negative', 'The Warper seams to be offline.');
-    });
-  },
-
-  alert: function(type, message) {
-    var validAlertTypes = ['progressive', 'negative', 'default'];
-    console.log(validAlertTypes.type);
-
-    if ($.inArray(validAlertTypes, type)) {
-      $('.alert-container').append('<div class="alert ' + type +'">' + message + '</div>');
-      setTimeout(function() {
-        $('.alert-container .alert').first().remove();
-      }, 6000);
-    } else {
-      console.log('invalid alert type');
-    }
-  }
-};
+// Leaflet integration for Map Warper (Wikimaps)
+
+
+var W_LEAFLET = {
+    map_div:		'map',
+    base_url:		'http://warper.wmflabs.org',
+    layers: 		{},
+    geo_search: 	true,
+    clustering: 	false,
+    bounds: 		false,
+    bounds_opacity: 0.5,
+    bounds_color: 	'blue',
+    bounds_weight: 	2,
+    markers: 		[],
+	init: function () {
+		
+		// set up the map
+		map = new L.Map(this.map_div);
+		this.map = map;
+
+		// create the base map layer 
+		var osmUrl='http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png';
+		var osmAttrib='Map data © <a href="http://openstreetmap.org">OpenStreetMap</a> contributors';
+		var osm = new L.TileLayer(osmUrl, {minZoom: 0, maxZoom: 20, attribution: osmAttrib});	
+		this.osm_layer = osm;	
+
+		// initial view
+		this.map.setView(new L.LatLng(60.1619343018,24.9516973282),12); // Helsinki
+		this.map.addLayer(osm);
+		
+		var clusterLayer = L.markerClusterGroup();
+		var featureLayer = L.layerGroup();
+		var boundsLayer = L.layerGroup();
+		
+		this.clusterLayer = clusterLayer;
+		this.featureLayer = featureLayer;
+		this.boundsLayer = boundsLayer;
+		
+		if (this.clustering) this.map.addLayer(this.clusterLayer);
+		this.map.addLayer(this.featureLayer);
+		if (this.bounds) this.map.addLayer(this.boundsLayer);
+
+		this.map.on('popupclose', function() {
+			W_LEAFLET.map.removeLayer(W_LEAFLET.boundsMarker);
+		});		
+	},
+
+	setView: function (lat, lon, zoom) {
```

Comment:
It is just a wrapper, so that ui code does not have to deal directly with Leaflet.